app/domain/authentication/authn_jwt/identity_providers/create_identity_provider.rb

```diff
@@ -61,11 +67,24 @@ def validate_identity_configuration
         end
 
         def multiple_identities_configured
-          identity_from_decoded_token_provider.identity_available? && identity_from_url_provider.identity_available?
+          identity_should_be_in_token? && identity_should_be_in_url?
         end
 
         def no_identities_configured
-          !identity_from_decoded_token_provider.identity_available? && !identity_from_url_provider.identity_available?
+          !identity_should_be_in_token? && !identity_should_be_in_url?
+        end
+
+        def identity_should_be_in_token?
```

Comment:
Please group the factory question and its object creation methods to be one after the other.